onnx/onnx.proto

```diff
@@ -212,6 +243,165 @@ message TensorProto {
 message SparseTensorProto {
   // The dimensions in the tensor.
   repeated int64 dims = 1;
+  // This field MUST be present this version of the IR.
   optional TensorProto indices = 2;
+  // This field MUST be present this version of the IR.  
   optional TensorProto values = 3;
 }
+
+// Define the types.
+message TypeProto {
+  message TensorTypeProto {
+  // This field MUST NOT have the value of UNDEFINED
+  // This field MUST be present for this version of the IR.
+    optional TensorProto.DataType elem_type = 1;
+  }
+
+  message SparseTensorTypeProto {
+  // This field MUST NOT have the value of UNDEFINED
+  // This field MUST be present for this version of the IR.
+    optional TensorProto.DataType elem_type = 1;
+  }
+
+  // ScalarType is needed to describe (at a minimum) scalar attribute values.
+  // When used in an attribute declaration, use the i|s|f fields a la the
+  // same encoding rules used by TensorProto.
+  message ScalarTypeProto {
```

Comment:
I don't think it's good to mix the scalars and sequences here with Tensor/SparseTensor types which are used in type inference. They are semantically different categories and are a bit confusing. For example, from seeing this at first I assumed that you propose introducing scalar types flowing through the graph which is already covered by 0-dim tensors